### PR TITLE
 validation schema | sign-up form

### PR DESCRIPTION
**Pull request by AlaaTaima**
Relates #15

### CODE REVIEW

File: client/src/components/pages/SignUpParent/index.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    const {
+      error,
+      email,
+      userName,
+      parentId,
+      password,
+      confirmPassword,
+    } = this.state;
+    signUpValidation
+      .isValid({ error, email, userName, parentId, password, confirmPassword })
```
Comment (IsraaSulaiman) [inline client/src/components/pages/SignUpParent/index.js]: Why do you pass error to the validation schema? And where do you check if password is equal to confirmPassword?

File: client/src/components/pages/SignUpParent/index.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          <Input
+            label="Your-Email"
+            name="email"
+            htmlFor="email"
+            id="email"
+            value={email}
+            type="email"
+            placeholder="example@gmail.com"
+            className="signUp__input"
+            onChange={({ target }) => this.setState({ email: target.value })}
```
Comment (IsraaSulaiman) [inline client/src/components/pages/SignUpParent/index.js]: It looks good this way. But I recommend you create a function to handle onChange events. Like this:
```javascript
 handleInputChange = ({target:{value}})=>{
   this.setState({email: value});
}
```
```suggestion
            onChange={this.handleInputChange}
```
Find the reason for my suggestion by the end of the article:
https://reactjs.org/docs/handling-events.html

File: client/src/components/pages/SignUpParent/index.js (hunk 1 above)
Comment (AlaaTaima) [reply client/src/components/pages/SignUpParent/index.js]: I haven't finished it yet.

File: client/src/components/pages/SignUpParent/index.js (hunk 2 above)
Comment (AlaaTaima) [reply client/src/components/pages/SignUpParent/index.js]: Thanks, I will see it.

File: client/src/components/pages/utils/signUpValidation.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    .required(),
+  password: yup
+    .string()
+    .min(8, 'Password has to be longer than 6 characters!')
+    .matches(/(?=.*[0-9])/, 'Password must contain a number.')
+    .required('Password is required!'),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref('password'), null], 'Passwords must match')
+
```
Comment (aminalakhsham) [inline client/src/components/pages/utils/signUpValidation.js]: remove this line

File: client/src/components/pages/SignUpParent/index.css [hunk 4]
```diff
@@ -0,0 +1,60 @@
+.signUp {
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: center;
+    font-family: "Roboto", sans-serif;
+}
```
Comment (asem1789) [inline client/src/components/pages/SignUpParent/index.css]: I run your code, 
you don't need all of these code under .signUp class

**Pull request merged**